tests/support/client.rs

```diff
@@ -29,6 +55,18 @@ pub fn http1<T: Into<String>>(addr: SocketAddr, auth: T) -> Client {
         Run::Http1 {
             absolute_uris: false,
         },
+        None,
+    )
+}
+
+pub fn http1_tls<T: Into<String>>(addr: SocketAddr, auth: T, tls: TlsConfig) -> Client {
```

Comment:
Any reason we're only testing TLS over http1? It could be helpful to add a similar `http2_tls` function.

Reply:
True, it will also be needed for the tap client as as its http2